### PR TITLE
Fix inferred substitution for irrelevant string terms

### PR DESCRIPTION
**Pull request by ajreynol**
Fixes https://github.com/cvc5/cvc5-projects/issues/659.

### CODE REVIEW

File: test/regress/cli/regress1/strings/proj-issue659.smt2 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+; COMMAND-LINE: -q
```
Comment (yoni206) [inline test/regress/cli/regress1/strings/proj-issue659.smt2:1]: why is this needed?

File: src/theory/strings/extf_solver.cpp [hunk 2]
```diff
@@ -735,7 +735,12 @@ Node ExtfSolver::getCurrentSubstitutionFor(int effort,
   if (effort >= 1 && n.getType().isStringLike())
   {
     Assert(effort < 3);
-    // normal forms
+    // Return self if the normal form has not been computed. This may happen
+    // for terms that are not relevant in the current context.
```
Comment (yoni206) [inline src/theory/strings/extf_solver.cpp:739]: would it make sense to assert that `n` is not relevant in the current context?

File: test/regress/cli/regress1/strings/proj-issue659.smt2 (hunk 1 above)
Comment (ajreynol) [reply test/regress/cli/regress1/strings/proj-issue659.smt2:1]: Since the regression involves partial functions whose model can't be checked.

File: src/theory/strings/extf_solver.cpp (hunk 2 above)
Comment (ajreynol) [reply src/theory/strings/extf_solver.cpp:739]: Its not trivial to check this :)

File: test/regress/cli/regress1/strings/proj-issue659.smt2 (hunk 1 above)
Comment (yoni206) [reply test/regress/cli/regress1/strings/proj-issue659.smt2:1]: So in that case maybe it is better to disable a tester rather than doing -q?

Comment (ajreynol) [reply test/regress/cli/regress1/strings/proj-issue659.smt2:1]: No, it is better to use `q` since this still (partially) checks the model is correct, i.e. nothing is "false" but unevaluated is permitted.

Comment (ajreynol) [reply test/regress/cli/regress1/strings/proj-issue659.smt2:1]: (FYI, `-q` is already used in 110 regressions)
